ironfish/src/utils/currency.ts

```diff
@@ -97,6 +97,18 @@ export class CurrencyUtils {
       throw e
     }
   }
+
+  static isValidOre(amount: string): boolean {
+    try {
+      const ore = this.decode(amount)
+      return ore >= MINIMUM_ORE_AMOUNT && ore <= MAXIMUM_ORE_AMOUNT
+    } catch (e) {
+      if (ErrorUtils.isNodeError(e) && e.code === 'NUMERIC_FAULT') {
+        return false
+      }
```

Comment:
Would `decode` throw these kinds of errors? It looks like `BigInt('0.5')` would throw a `RangeError`, which this wouldn't catch.